docs/source/python/extending_types.rst

```diff
@@ -46,6 +46,11 @@ The ``__arrow_array__`` method takes an optional `type` keyword which is passed
 through from :func:`pyarrow.array`. The method is allowed to return either
 a :class:`~pyarrow.Array` or a :class:`~pyarrow.ChunkedArray`.
 
+.. note::
+
+    For a lower-level API that integrates with libraries other than PyArrow,
+    consider the :doc:`/format/CDataInterface/PyCapsuleInterface`.
```

Comment:
I'm not sure "lower-level" is really relevant here, as it's a protocol and not a public-facing API.
It's more useful to point out that the PyCapsule-based interface is more general and does not imply a dependency on PyArrow.

Reply:
Reworded this. Thanks for the suggestion.